object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedObjectStoreWithFilter.java

```diff
@@ -0,0 +1,65 @@
+package org.hypertrace.config.objectstore;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.hypertrace.config.service.change.event.api.ConfigChangeEventGenerator;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+/**
+ * IdentifiedObjectStoreWithFilter is an abstraction over {@link IdentifiedObjectStore} to allow
+ * filtering on fetched configs
+ *
+ * @param <T> Config Object
+ * @param <F> Filter
+ */
+public abstract class IdentifiedObjectStoreWithFilter<T, F> extends IdentifiedObjectStore<T> {
+  protected IdentifiedObjectStoreWithFilter(
+      ConfigServiceGrpc.ConfigServiceBlockingStub configServiceBlockingStub,
+      String resourceNamespace,
+      String resourceName,
+      ConfigChangeEventGenerator configChangeEventGenerator) {
+    super(configServiceBlockingStub, resourceNamespace, resourceName, configChangeEventGenerator);
+  }
+
+  public List<T> getAllFilteredConfigs(RequestContext context, F filter) {
+    Stream<T> configObjectsStream = getAllObjects(context).stream().map(ConfigObject::getData);
+    if (!isFilterEmpty(filter)) {
+      configObjectsStream =
+          configObjectsStream
+              .map(configObject -> filterConfig(configObject, filter))
+              .filter(Optional::isPresent)
```

Comment:
1. filter + map here is equivalent to `flatMap(Optional::stream)`
2. we should be matching the non-filtered APIs (both in terms of name as well as signature), returning the wrapper objects. If we think it's useful to expose either, we could expose a second shortcut method.

Reply:
1 - yes will fix
2 - didn't understand.. do u mean the contextualWrapped objects should be returned instead? hmm was considering that but didn't really see many such usages.. but yeah, should keep APIs consistent - will fix

Reply:
fixed